### ISSUE TITLE
Color control without value throws error

### ISSUE THREAD

**Issue by blowsie** [storybookjs/storybook#14085]
**Describe the bug**
After following the guide from the blog.
https://storybook.js.org/blog/storybook-vue3/

I click on the color picker and the web page throws an error

**To Reproduce**

* Install example from here https://storybook.js.org/blog/storybook-vue3/
* Run Storybook
* View button component, click color picker in controls.

**Expected behavior**
color picker to work

**Screenshots**
![3YJIlcjyzG](https://user-images.githubusercontent.com/308572/109493029-cd622d80-7a83-11eb-9c36-53a1362d7bda.gif)

**Code snippets**
n/a

**System**

```
Environment Info:

  System:
    OS: Windows 10 10.0.18363
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 15.10.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 7.5.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 88.0.4324.190
    Edge: Spartan (44.18362.449.0)
  npmPackages:
    @storybook/addon-actions: ^6.2.0-beta.4 => 6.2.0-beta.4
    @storybook/addon-essentials: ^6.2.0-beta.4 => 6.2.0-beta.4
    @storybook/addon-links: ^6.2.0-beta.4 => 6.2.0-beta.4
    @storybook/addon-postcss: ^2.0.0 => 2.0.0
    @storybook/vue3: ^6.2.0-beta.4 => 6.2.0-beta.4
```

**Additional context**

```
The above error occurred in the <ColorPicker> component:
    in ColorPicker (created by ColorControl)
    in div (created by Context.Consumer)
    in Styled(div) (created by ColorControl)
    in button (created by Context.Consumer)
    in Styled(button) (created by ForwardRef)
    in ForwardRef (created by ForwardRef)
    in ForwardRef (created by Context.Consumer)
    in Styled(Component) (created by Button)
    in Button (created by Context.Consumer)
    in Styled(Button) (created by ColorControl)
    in ColorControl (created by ColorControl)
    in Suspense (created by ColorControl)
    in ColorControl (created by ArgControl)
    in ArgControl (created by ArgRow)
    in td (created by ArgRow)
    in tr (created by ArgRow)
    in ArgRow (created by ArgsTable)
    in SectionRow (created by ArgsTable)
    in tbody (created by ArgsTable)
    in table (created by Context.Consumer)
    in Styled(table) (created by ArgsTable)
    in div (created by Context.Consumer)
    in Styled(div) (created by ArgsTable)
    in ArgsTable (created by ControlsPanel)
    in ControlsPanel
    in div (created by AddonPanel)
    in AddonPanel
    in div (created by Context.Consumer)
    in Styled(div)
    in div (created by Context.Consumer)
    in Styled(div)
    in Unknown
    in Unknown
    in Unknown
    in Unknown (created by ManagerConsumer)
    in ManagerConsumer (created by Panel)
    in Panel (created by Layout)
    in div (created by Context.Consumer)
    in Styled(div) (created by Panel)
    in Panel (created by Layout)
    in div (created by Context.Consumer)
    in Styled(div) (created by Main)
    in div (created by Context.Consumer)
    in Styled(div) (created by Main)
    in Main (created by Layout)
    in Layout (created by Context.Consumer)
    in WithTheme(Layout)
    in Unknown
    in div (created by Context.Consumer)
    in Styled(div)
    in Unknown
    in Unknown (created by SizeMeRenderer(Component))
    in SizeMeReferenceWrapper (created by SizeMeRenderer(Component))
    in SizeMeRenderer(Component) (created by SizeMe(Component))
    in SizeMe(Component)
    in ThemeProvider
    in Unknown (created by ManagerConsumer)
    in ManagerConsumer (created by Manager)
    in EffectOnMount (created by Manager)
    in Manager (created by Context.Consumer)
    in Location (created by QueryLocation)
    in QueryLocation (created by Root)
    in LocationProvider (created by Root)
    in HelmetProvider (created by Root)
    in Root

React will try to recreate this component tree from scratch using the error boundary you provided, LocationProvider.
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1199622255231368/1200025950355960) by [Unito](https://www.unito.io/learn-more)

**Comment by pocka** [comment]
Confirmed in v6.2.0-beta.10, and happened in Storybook for React (same version, CRA) too. Clicking an empty color picker crashes the entire Storybook UI (you can check it with stories created by `npx sb init`). Also tested on stable (v6.1.21) but did not happen in v6.1.21.

<details>
<summary>Error stack</summary>

```
Uncaught TypeError: data is null
    toState http://localhost:6006/2.407e2ae41837c1278f16.bundle.js:4229
    ColorPicker http://localhost:6006/2.407e2ae41837c1278f16.bundle.js:4324
    constructClassInstance http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:100824
    updateClassComponent http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:105533
    beginWork http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:107181
    callCallback http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:92053
    invokeGuardedCallbackDev http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:92102
    invokeGuardedCallback http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:92164
    beginWork$1 http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:112067
    performUnitOfWork http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:110882
    workLoopSync http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:110810
    renderRootSync http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:110773
    performSyncWorkOnRoot http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:110396
    flushSyncCallbackQueueImpl http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:99435
    unstable_runWithPriority http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:121560
    runWithPriority$1 http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:99384
    flushSyncCallbackQueueImpl http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:99430
    flushSyncCallbackQueue http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:99417
    discreteUpdates$1 http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:110523
    discreteUpdates http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:91864
    dispatchDiscreteEvent http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:93997
    addEventBubbleListener http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:94119
    addTrappedEventListener http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:96519
    listenToNativeEvent http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:96482
    listenToAllSupportedEvents http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:96434
    listenToAllSupportedEvents http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:96432
    createRootImpl http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:113990
    ReactDOMBlockingRoot http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:113938
    createLegacyRoot http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:114003
    legacyCreateRootFromDOMContainer http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:114084
    legacyRenderSubtreeIntoContainer http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:114110
    render http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:114206
    renderStorybookUI http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:34904
    node_modules storybook/core-client/dist/esm/manager/index.js/<@http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:35353
    promise callback*./node_modules/ storybook/core-client/dist/esm/manager/index.js@http://localhost:6006/vendors~main.1134f8041fe517ffe802.bundle.js:35351
    __webpack_require__ http://localhost:6006/runtime~main.acf0b383531550796e24.bundle.js:85
    0 http://localhost:6006/main.198ac76f08986c45c37b.bundle.js:33
    __webpack_require__ http://localhost:6006/runtime~main.acf0b383531550796e24.bundle.js:85
    checkDeferredModules http://localhost:6006/runtime~main.acf0b383531550796e24.bundle.js:46
    webpackJsonpCallback http://localhost:6006/runtime~main.acf0b383531550796e24.bundle.js:33
    <anonymous> http://localhost:6006/main.198ac76f08986c45c37b.bundle.js:1
```

</details>

**Comment by pocka** [comment]
@shilman 
I changed the title because it turns out this is `addon-controls`' (or Storybook UI's?) bug instead of Vue specific one. Could you take a look at this?

**Comment by shilman** [comment]
Thanks @pocka, we'll take care of it! 🙏

**Comment by ghengeveld** [comment]
Turns out the issue isn't if there's *no value*, it's when the value is `null`.

**Comment by ghengeveld** [comment]
Resolved in #14316 and released in 6.2.0-rc.9
